dbt/project.py

```diff
@@ -25,6 +25,15 @@
 
 default_active_profiles = ['user']
 
+class ProjectConfigError(Exception):
```

Comment:
I think this error type is too narrowly scoped - it's generally a bad sign if an Error/Exception class is only useful in one specific situation.  I'm generally wary of creating new Error/Exception types, but if we're going to do it, I think something like this is more versatile:

```python
class DbtProjectError(Exception):
  def __init__(self, message, project):
    self.project = project
    super(DbtProjectError, self).__init__(message)

...
  if key not in target_cfg or len(str(target_cfg[key])) == 0:
    raise DbtProjectError("Expected project configuration '{}' was not supplied".format(key), self)
...

except DbtProjectError as e:
  print e.message
  log.debug(e.project)
```